### PR TITLE
Feature web observe appointment

### PR DESCRIPTION
**Pull request by Markovic-Filip**

### CODE REVIEW

**Review by st3fan7 — DISMISSED** [review]

**Review by st3fan7 — CHANGES_REQUESTED** [review]
Please resolve the conflicts.

File: WebApplication/Controller/AppointmentController.cs [hunk 1]
```diff
@@ -25,6 +25,47 @@ public AppointmentController(IAppointmentService appointmentService, IMedicalRec
...
+        /*
+        [HttpGet] //GET api/Appointment
+        public IActionResult Get()
+        {
+            List<Appointment> appointments = _appointmentService.GetAppointmentsByPatientId("2406978890046");
+            List<GetAppointmentDTO> appointmentsDTO = AppointmentAdapter.ListAppointmentToListGetAppointmentDTO(appointments);
+            return Ok(appointmentsDTO);
+        }*/
+
+        [HttpGet("allSurveyableAppointments")] //GET api/Appointment
```
Comment (jelena-vlajkov) [inline WebApplication/Controller/AppointmentController.cs]: Delete commented lines.

File: Backend/Schedules/Repository/IAppointmentRepository.cs [hunk 2]
```diff
@@ -15,7 +15,9 @@ public interface IAppointmentRepository : ICreate<Appointment>, IGet<Appointment
    {
         Dictionary<int, Appointment> GetAppointmentsBy(DateTime date);
         Dictionary<int, Appointment> GetScheduledFromToday();
-
+        Appointment getAppointmentById(int id);
+        Appointment update(Appointment appointment);
+        IEnumerable<Appointment> GetAppointmentsByPatientId(string Id);
         IEnumerable<Appointment> GetBy(string doctorId, DateTime date);
     }
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Repository/IAppointmentRepository.cs:20]: Please follow the convention. 
getAppointmentById -> GetAppointmentById, etc.

File: Backend/Schedules/Repository/IAppointmentRepository.cs [hunk 3]
```diff
@@ -15,7 +15,9 @@ public interface IAppointmentRepository : ICreate<Appointment>, IGet<Appointment
    {
         Dictionary<int, Appointment> GetAppointmentsBy(DateTime date);
         Dictionary<int, Appointment> GetScheduledFromToday();
-
+        Appointment getAppointmentById(int id);
+        Appointment update(Appointment appointment);
+        IEnumerable<Appointment> GetAppointmentsByPatientId(string Id);
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Repository/IAppointmentRepository.cs:18]: On the other hand, every sql repository will inherit MySqlRepository class and in that class you can find methods like Update and GetObject, exact same methods you wanted to implement here.

File: Backend/Schedules/Repository/MySqlRepository/AppointmentSqlRepository.cs [hunk 4]
```diff
@@ -13,11 +13,31 @@ public class AppointmentSqlRepository : MySqlrepository<Appointment, int>,
         IAppointmentRepository
     {
         public AppointmentSqlRepository(MySqlContext context) : base(context) { }
+
+        public Appointment getAppointmentById(int id)
+        {
+            return GetAll().SingleOrDefault(entity => entity.Id.Equals(id));
+        }
+
         public Dictionary<int, Appointment> GetAppointmentsBy(DateTime date)
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Repository/MySqlRepository/AppointmentSqlRepository.cs:17]: Useless method, base class already has GetObject.

File: Backend/Schedules/Repository/MySqlRepository/AppointmentSqlRepository.cs [hunk 5]
```diff
@@ -27,5 +47,10 @@ public IEnumerable<Appointment> GetBy(string doctorId, DateTime date)
         {
             return GetAll().ToList().Where(a => a.Period.StartTime.CompareTo(DateTime.Now) <= 0).ToDictionary(a => a.Id);
         }
+
+        public Appointment update(Appointment appointment)
+        {
+            return base.Update(appointment);
+        }
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Repository/MySqlRepository/AppointmentSqlRepository.cs]: You just returned base class Update, so you already have a function you need.

File: Backend/Schedules/Service/AppointmentService.cs [hunk 6]
```diff
@@ -7,22 +7,82 @@
...
+        {
+            return _appointmentRepository.GetAppointmentsByPatientId(id).ToList();
+        }
+
+        public List<Appointment> GetCancelableAppointments(string id)
+        {
+            List<Appointment> appointments = _appointmentRepository.GetAppointmentsByPatientId(id).ToList();
+            List<Appointment> cancelableAppointments = new List<Appointment>();
+            cancelableAppointments = appointments.Where(p => !(p.CanceledByPatient) && !(p.Finished) && ((p.Start - DateTime.Now).TotalDays > 2.00)).ToList();
+
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Service/AppointmentService.cs:60]: **Magic number 2.00**. Change that!

**Review by jelena-vlajkov — CHANGES_REQUESTED** [review]
Resolve conflicts before you make a PR + change the things I marked.

File: Backend/Schedules/Service/AppointmentService.cs [hunk 7]
```diff
@@ -7,21 +7,84 @@
...
+            List<Appointment> surveyableAppointments = new List<Appointment>();
+            surveyableAppointments = appointments.Where(p => !surveys.Any(l => p.Id == l.AppointmentId) && p.Finished == true).ToList();
+            
+            return surveyableAppointments;
+        }
+
+        public bool UpdateCanceled(int appointmentId)
+        {
+            Appointment appointment = _appointmentRepository.GetObject(appointmentId);
+            if (appointment == null) {
```
Comment (danica1998) [inline Backend/Schedules/Service/AppointmentService.cs:77]: you can join those two if statements into one

**Review by danica1998 — APPROVED** [review]

**Pull request closed without merging**